SwiftUIDZ/SwiftUIDZ/CustomTabBarView.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  CustomTabBarView.swift
+//  SwiftUIDZ
+//
+//  Created by Анна Сычева on 21.01.2023.
+//
+
+import SwiftUI
+
+/// Кастомный таббар
+struct CustomTabBarView: View {
+    
+    // MARK: - Constants
+    
+    private enum Constants {
+        static let duration = 0.2
+        static let imageSize: CGFloat = 30
+        static let circleSize: CGFloat = 25
+        static let offset: CGFloat = 40
+        static let paddingBottom: CGFloat = 10
+        static let offsetMultiplay = 1.02
+    }
+    
+    // MARK: - Public Properties
+    
+    @Binding var currentTab: TabType
+    
+    // MARK: - Body
+    
+    var body: some View {
+        tabIconWithCircle
+    }
+    
+    // MARK: - Private Properties
+    
+    private var tabIconWithCircle: some View {
+        GeometryReader { proxy in
+            let width = proxy.size.width
+            HStack {
+                ForEach(TabType.allCases, id: \.rawValue) { tab in
+                    Button {
+                        withAnimation(.easeInOut(duration: Constants.duration)) {
+                            currentTab = tab
+                        }
+                    } label: {
+                        Image(systemName: tab.rawValue)
+                            .renderingMode(.template)
+                            .resizable()
+                            .aspectRatio(contentMode: .fit)
+                            .frame(width: Constants.imageSize, height: Constants.imageSize)
+                            .frame(maxWidth: .infinity)
+                            .foregroundColor(currentTab == tab ? .blue : .gray)
+                    }
+                }
+            }
+            .frame(maxWidth: .infinity)
+            .background(alignment: .leading) {
+                Circle()
```

Comment:
```suggestion
                Circle()
                .fill(Color.yellow)
                     .frame(width: Constants.circleSize, height: Constants.circleSize)
                     .offset(x: Constants.offset)
                     .offset(x: indicatorOffset(width: width * Constants.offsetMultiplay))
```
Вынеси в вычисляемое

Reply:
не смогла вынести из-за этого
        GeometryReader { proxy in
            let width = proxy.size.width